pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -202,6 +199,39 @@ func (s *Service) createLaunchTemplateData(scope *scope.MachinePoolScope, userDa
 	}
 	s.scope.Info("Security Groups", "security groups", data.SecurityGroupIds)
 
+	// Pick image from the machinepool configuration, or use a default one.
+	if scope.AWSMachinePool.Spec.AWSLaunchTemplate.AMI.ID != nil { // nolint:nestif
```

Comment:
from my mini refactor, there's an `lt` you can use instead of `scope.AWSMachinePool.Spec.AWSLaunchTemplate`

Reply:
oh right. i saw that `lt` when i rebased. will update